addons/l10n_th/models/res_partner.py

```diff
@@ -14,4 +16,10 @@ def _compute_l10n_th_branch_name(self):
                 partner.l10n_th_branch_name = ""
             else:
                 code = partner.company_registry
-                partner.l10n_th_branch_name = f"Branch {code}" if code else "Headquarter"
+                partner.l10n_th_branch_name = f"Branch {code}" if code and code != "00000" else "Headquarter"
+
+    @api.constrains('company_registry')
+    def _check_company_registry_l10n_th(self):
+        for partner in self:
+            if partner.country_code == "TH" and partner.company_registry and not re.fullmatch(r'\d{5}', partner.company_registry):
+                raise ValidationError(_("Branch Code must be exactly 5 digits."))
```

Comment:
I modern versions, you must translate using the env.


```suggestion
                raise ValidationError(partner.env._("The branch Code must be exactly 5 digits."))
```

Reply:
Noted! :saluting_face: